src/com/webonise/carbuilder/CompanyFactory.java

```diff
@@ -0,0 +1,40 @@
+package com.webonise.carbuilder;
+
+public class CompanyFactory extends AbstractCarFactory{
+
+	@Override
+	public ITypeOfCar getTypeOfCar(String TypeOfCar) {
+		// TODO Auto-generated method stub
+		return null;
+	}
+
+	@Override
+	public ICompany getCompany(String Company) {
+		if(Company == null){
+	         return null;
+	      }		
```

Comment:
Please correct indentation.
As i suggested before on slack u can create a macro in intellij and save time for reformating code and optimizing imports. Here is a link which u can follow
http://stackoverflow.com/questions/946993/intellij-reformat-on-file-save